packages/react-ai/src/components/AIConversation/views/default/Form.tsx

```diff
@@ -62,8 +66,46 @@ export const Form: Required<ControlsContextProps>['Form'] = ({
   const [composing, setComposing] = React.useState(false);
   const isInputEmpty = !input?.text?.length && !input?.files?.length;
 
+  const validateAttachments = React.useCallback(
+    async (files: File[]) => {
+      const previousFiles = input?.files ?? [];
+      const { acceptedFiles, hasMaxError, hasMaxSizeError } =
+        await attachmentsValidator({
+          files: [...files, ...previousFiles],
+          maxAttachments,
+          maxAttachmentSize,
+        });
+
+      if (hasMaxError ?? hasMaxSizeError) {
+        let error = '';
+        if (hasMaxError) {
+          error += displayText.getMaxAttachmentErrorText(maxAttachments);
+        }
+        if (hasMaxSizeError) {
+          error += displayText.getAttachmentSizeErrorText(
+            // base64 size is about 137% that of the file size
+            // https://en.wikipedia.org/wiki/Base64#MIME
+            humanFileSize((maxAttachmentSize - 814) / 1.37, true)
+          );
```

Comment:
Looks like we could avoid passing displayText as a prop if we handle the validation logic in the parent and just pass an `onValidate` callback to the `Form`?

Reply:
Just noting that if displayText is not being provided to override components not concerned abt the logic living here fir the immediate term

Reply:
Good call, moved this up and passed an onValidate callback